pkg/podidentitycontroller/controller.go

```diff
@@ -0,0 +1,337 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package podidentitycontroller
+
+import (
+	"errors"
+	"fmt"
+	"time"
+
+	myspec "github.com/m3db/m3db-operator/pkg/apis/m3dboperator/v1"
+	clientset "github.com/m3db/m3db-operator/pkg/client/clientset/versioned"
+	samplescheme "github.com/m3db/m3db-operator/pkg/client/clientset/versioned/scheme"
+	clusterlisters "github.com/m3db/m3db-operator/pkg/client/listers/m3dboperator/v1"
+	"github.com/m3db/m3db-operator/pkg/k8sops"
+	"github.com/m3db/m3db-operator/pkg/k8sops/labels"
+	"github.com/m3db/m3db-operator/pkg/k8sops/podidentity"
+	"github.com/m3db/m3db-operator/pkg/util/eventer"
+
+	corev1 "k8s.io/api/core/v1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/util/clock"
+	"k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	appslisters "k8s.io/client-go/listers/apps/v1"
+	corelisters "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+
+	"github.com/uber-go/tally"
+	"go.uber.org/zap"
+)
+
+const (
+	controllerName = "m3db-podidentity-controller"
+	workQueueName  = "m3db-pod-identity"
+)
+
+var (
+	errOrphanedPod = errors.New("pod does not belong to an m3db cluster")
+)
+
+// Controller object.
+type Controller struct {
```

Comment:
What did we decide we were going to do about this extra controller for now? Whatever it is, in the meantime can we make an issue to make the Controller struct reusable so our repo can be DRY lol

Reply:
Yeah you're spot on here, this is a significantly better idea 😂 . Ripped out a ton of the boiler plate in exchange for 2 work queues in the main controller, we can always split out later if it gets crowded in that code and we invest in better controller construction.